one_fm/overrides/payroll_entry.py

```diff
@@ -0,0 +1,311 @@
+			department=self.department,
+			designation=self.designation,
+			grade=self.grade,
+			currency=self.currency,
+			start_date=self.start_date,
+			end_date=self.end_date,
+			payroll_payable_account=self.payroll_payable_account,
+			salary_slip_based_on_timesheet=self.salary_slip_based_on_timesheet,
+			project=project,
+			custom_payroll_cycle_start_date=getdate(self.start_date).day,
+			custom_payroll_cycle_end_date=getdate(self.end_date).day,
+			payroll_type=self.payroll_type
+		)
+
+		if not self.salary_slip_based_on_timesheet:
+			filters.update(dict(payroll_frequency=self.payroll_frequency))
+
+		return filters
+
+@frappe.whitelist()
+def get_start_end_dates(payroll_frequency, start_date=None, company=None):
+	"""Returns dict of start and end dates for given payroll frequency based on start_date"""
+	start_date, end_date = None, None
+	if payroll_frequency == "Monthly":
+		#fetch Payroll date's day
+		date = frappe.db.get_single_value('HR and Payroll Additional Settings', 'payroll_date')
+		year = getdate().year - 1 if getdate().day < cint(date) and  getdate().month == 1 else getdate().year
+		month = getdate().month if getdate().day >= cint(date) else getdate().month - 1
+
+		#calculate Payroll date, start and end date.
+		payroll_date = datetime(year, month, cint(date)).strftime("%Y-%m-%d")
+		start_date = add_to_date(payroll_date, months=-1)
+		end_date = add_to_date(payroll_date, days=-1)
+
+	return frappe._dict({"start_date": start_date, "end_date": end_date})
+
+
+def get_employee_list(
+	filters: frappe._dict,
+	searchfield=None,
+	search_string=None,
+	fields: list[str] | None = None,
+	as_dict=True,
+	limit=None,
+	offset=None,
+	ignore_match_conditions=False,
+) -> list:
+	sal_struct = get_salary_structure(
+		filters.company,
+		filters.currency,
+		filters.salary_slip_based_on_timesheet,
+		filters.payroll_frequency,
+	)
+
+	if not sal_struct:
+		return []
+
+	emp_list = get_filtered_employees(
+		sal_struct,
+		filters,
+		searchfield,
+		search_string,
+		fields,
+		as_dict=as_dict,
+		limit=limit,
+		offset=offset,
+		ignore_match_conditions=ignore_match_conditions,
+	)
+
+	if as_dict:
+		employees_to_check = {emp.employee: emp for emp in emp_list}
+	else:
+		employees_to_check = {emp[0]: emp for emp in emp_list}
+
+	return remove_payrolled_employees(employees_to_check, filters.start_date, filters.end_date)
+
+
+
+def get_filtered_employees(
+	sal_struct,
+	filters,
+	searchfield=None,
+	search_string=None,
+	fields=None,
+	as_dict=False,
+	limit=None,
+	offset=None,
+	ignore_match_conditions=False,
+) -> list:
+	SalaryStructureAssignment = frappe.qb.DocType("Salary Structure Assignment")
+	Employee = frappe.qb.DocType("Employee")
+	Attendance = frappe.qb.DocType("Attendance")
+
+	query = (
+		frappe.qb.from_(Employee)
+		.join(SalaryStructureAssignment)
+		.on(Employee.name == SalaryStructureAssignment.employee)
+		.where(
+			(SalaryStructureAssignment.docstatus == 1)
+			& (Employee.status != "Inactive")

```

Comment:
On the employee doctype, there is no inactive status

just active, courtcase, left, absconding

Reply:
@AbrahamAdekunle Nice catch. Good work.